python/tank/util/shotgun.py

```diff
@@ -151,67 +158,78 @@ def __get_sg_config_data(shotgun_cfg_path, user="default"):
-    if "api_script" not in config_data:
-        raise TankError("Missing required field 'api_script' in config '%s'" % shotgun_cfg_path)
-    if "api_key" not in config_data:
-        raise TankError("Missing required field 'api_key' in config '%s'" % shotgun_cfg_path)
-    
+
+    def _raise_missing_key(key):
+        raise TankError(
+            "Missing required field '%s' in config '%s' for script user authentication." % (key, shotgun_cfg_path)
+        )
+
+    if not config_data.get("host"):
+        _raise_missing_key("host")
+
+    # The script authentication credentials need to be complete in order to work. They can be completely
+    # omitted or fully specified, but not halfway configured.
+    if config_data.get("api_script") and not config_data.get("api_key"):
+        _raise_missing_key("api_key")
+    if not config_data.get("api_script") and config_data.get("api_key"):
+        _raise_missing_key("api_script")
+
     return config_data
 
-def __create_sg_connection(shotgun_cfg_path, evaluate_script_user, user="default"):
-    """
-    Creates a standard toolkit shotgun connection.
 
-    :param shotgun_cfg_path: path to a configuration file to read settings from
-    :param evaluate_script_user: if True, the id of the script user will be 
-                                 looked up and returned.
-    :param user: If a multi-user config is used, this is the user to create the connection for.
-    
-    :returns: tuple with (sg_api_instance, script_user_dict) where script_user_dict is None if
-              evaluate_script_user is False else a dictionary with type and id keys. 
+def __create_sg_connection(config_data=None):
+    """
+    Creates a standard Toolkit shotgun connection.
+
+    :param config_data: Configuration data dictionary. Keys host, api_script and api_key are
+                        expected, while http_proxy is optional. If None, the authentication module
+                        will be responsible for determining which credentials to use. This parameter
+                        is present for legacy reason to support a Psyop workflow. You shouldn't
+                        be using this.
+    :returns: A Shotgun connection.
     """
 
-    # get connection parameters
-    config_data = __get_sg_config_data(shotgun_cfg_path, user)
-
-    # create API
-    sg = Shotgun(config_data["host"],
-                 config_data["api_script"],
-                 config_data["api_key"],
-                 http_proxy=config_data.get("http_proxy", None))
+    if config_data:
+        # Credentials were passed in, so let's run the legacy authentication
+        # mechanism for script user.
+        sg = shotgun_api3.Shotgun(
+            config_data["host"],
+            script_name=config_data["api_script"], api_key=config_data["api_key"],
+            http_proxy=config_data.get("http_proxy")
+        )
+    else:
+        from .. import api
```

Comment:
if this is to avoid cyclic imports, maybe better to have it at the top of the method? (with a comment)